### PR TITLE
Door Spam

### PR DESCRIPTION
**Pull request by playday3008**

### CODE REVIEW

**Comment by playday3008** [comment]
And here 2 commits bcoz i forgot how to correctly fork master brunch to my new brunch

**Comment by dapaster** [comment]
isDefusing return

**Comment by playday3008** [comment]
> isDefusing return

fixed, thanks

**Comment by newcommerdontblame** [comment]
isnt it repost of notgoodusename's pr?

**Comment by deboogergithub** [comment]
@notgoodusename

**Comment by deboogergithub** [comment]
its not a door spammer, its an "e" macro

**Comment by playday3008** [comment]
> isnt it repost of notgoodusename's pr?

Not repost, but inspired by him pr

**Comment by notgoodusename** [comment]
use traceray to check if it is a door

**Comment by playday3008** [comment]
How can I did it?

**Comment by notgoodusename** [comment]
look at how fast plant is done

**Comment by tirziz** [comment]
Im sorry but, why is this useful at all lol

**Comment by notgoodusename** [comment]
nice

**Comment by unnamed01X** [comment]
while Door spam is activated, then, when im trying to press E, to open the doors, the game is getting crashed.

**Comment by playday3008** [comment]
> while Door spam is activated, then, when im trying to press E, to open the doors, the game is getting crashed.

Check in debug and show here, bcoz I can't (probably my potato PC can't 😭)

**Comment by SapphireCodenz** [comment]
playday is gay

**Comment by playday3008** [comment]
> while Door spam is activated, then, when im trying to press E, to open the doors, the game is getting crashed.

@unnamed01X fixed

**Comment by unnamed01X** [comment]
> > while Door spam is activated, then, when im trying to press E, to open the doors, the game is getting crashed.
> 
> @unnamed01X fixed

i'll try, thanks

File: Osiris/Hacks/Misc.cpp [hunk 1]
```diff
@@ -791,3 +791,23 @@ void Misc::purchaseList(GameEvent* event) noexcept
...
 }
+
+void Misc::doorSpam(UserCmd* cmd) noexcept {
+
+    if (!localPlayer || !config->misc.doorSpam || localPlayer->isDefusing())
+        return;
+
+    static bool doorSpam = true;
+    constexpr float doorRange{ 200.0f };
+    Vector viewAngles{ cos(degreesToRadians(cmd->viewangles.x)) * cos(degreesToRadians(cmd->viewangles.y)) * doorRange,
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:802]: Use Vector::fromAngles()

**Comment by unnamed01X** [comment]
> > while Door spam is activated, then, when im trying to press E, to open the doors, the game is getting crashed.
> 
> @unnamed01X fixed

hey, there's something wrong with that option. The debuger doesnt shows anything. I just release that when i activated this "door spam" option, then when im trying to open a door, the game is getting crashed. What's going on?

File: Osiris/Hacks/Misc.cpp (hunk 1 above)
Comment (playday3008) [reply Osiris/Hacks/Misc.cpp:802]: How? Because IDK, sry

Comment (playday3008) [reply Osiris/Hacks/Misc.cpp:802]: Can you show me example?

Comment (danielkrupinski) [reply Osiris/Hacks/Misc.cpp:802]: https://github.com/danielkrupinski/Osiris/blob/92ec15dc50a745e043c1e2d3a77faf9e83e141f4/Osiris/Hacks/Misc.cpp#L302

**Comment by danielkrupinski** [comment]
It shouldn't do anything if 'USE' button was pressed and quickly released (to open door normally).

File: Osiris/Hacks/Misc.cpp [hunk 2]
```diff
@@ -807,6 +807,26 @@ void Misc::purchaseList(GameEvent* event) noexcept
...
+
+    Trace trace;
+    const auto startPos = localPlayer->getEyePosition();
+    const auto endPos = startPos + Vector::fromAngle(cmd->viewangles) * doorRange;
+    interfaces->engineTrace->traceRay({ startPos, endPos }, 0x46004009, localPlayer.get(), trace);
+
+    if (trace.entity && trace.entity->getClientClass()->classId == ClassId::PropDoorRotating)
+        if (cmd->buttons & UserCmd::IN_USE) {
+            doorSpam ? cmd->buttons |= UserCmd::IN_USE : cmd->buttons &= ~UserCmd::IN_USE;
+            doorSpam = !doorSpam;
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:826]:
```cpp
if (cmd->buttons & UserCmd::IN_USE && cmd->tickCount & 1)
    cmd->buttons &= ~UserCmd::IN_USE;
```
and remove `doorSpam` variable

**Review by AstroOrbis — APPROVED** [review]

**Pull request closed without merging**